packages/@aws-cdk/aws-vpcv2-alpha/lib/route.ts

```diff
@@ -0,0 +1,582 @@
+  readonly destination: string;
+
+  /**
+   * The gateway or endpoint targeted by the route.
+   */
+  readonly target: RouteTargetType;
+}
+
+/**
+ * Properties to define a route.
+ */
+export interface RouteProps {
+  /**
+   * The ID of the route table for the route.
+   * @attribute routeTable
+   */
+  readonly routeTable: IRouteTable;
+
+  /**
+   * The IPv4 or IPv6 CIDR block used for the destination match.
+   *
+   * Routing decisions are based on the most specific match.
+   */
+  readonly destination: string;
+
+  /**
+   * The gateway or endpoint targeted by the route.
+   */
+  readonly target: RouteTargetType;
+
+  /**
+   * The resource name of the route.
+   * @default none
+   */
+  readonly routeName?: string;
+}
+
+/**
+ * Creates a new route with added functionality.
+ * @resource AWS::EC2::Route
+ */
+export class Route extends Resource implements IRouteV2 {
+  /**
+   * The IPv4 or IPv6 CIDR block used for the destination match.
+   *
+   * Routing decisions are based on the most specific match.
+   */
+  public readonly destination: string;
+
+  /**
+   * The gateway or endpoint targeted by the route.
+   */
+  public readonly target: RouteTargetType;
+
+  /**
+   * The route table for the route.
+   * @attribute routeTable
+   */
+  public readonly routeTable: IRouteTable;
+
+  /**
+   * The type of router the route is targetting
+   */
+  public readonly targetRouterType: RouterType
+
+  /**
+   * The route CFN resource.
+   */
+  public readonly resource?: CfnRoute;
+
+  constructor(scope: Construct, id: string, props: RouteProps) {
+    super(scope, id);
+
+    this.destination = props.destination;
+    this.target = props.target;
+    this.routeTable = props.routeTable;
+
+    this.targetRouterType = this.target.gateway ? this.target.gateway.routerType : RouterType.VPC_ENDPOINT;
+
+    // Gateway generates route automatically via its RouteTable, thus we don't need to generate the resource for it
+    if (!(this.target.endpoint instanceof GatewayVpcEndpoint)) {
+      this.resource = new CfnRoute(this, 'Route', {
+        routeTableId: this.routeTable.routeTableId,
+        destinationCidrBlock: this.destination,
+        destinationIpv6CidrBlock: this.destination,
+        [routerTypeToPropName(this.targetRouterType)]: this.target.gateway ? this.target.gateway.routerTargetId :
+          this.target.endpoint ? this.target.endpoint.vpcEndpointId : null,
+      });
+    }
+
+    if (this.targetRouterType == RouterType.GATEWAY) {
+      if (this.target.gateway instanceof InternetGateway) {
+        new CfnVPCGatewayAttachment(this, 'GWAttachment', {
+          vpcId: this.target.gateway.vpcId,
+          internetGatewayId: this.target.gateway.routerTargetId,
+        });
+      } else if (this.target.gateway instanceof VPNGateway) {
+        new CfnVPCGatewayAttachment(this, 'GWAttachment', {
+          vpcId: this.target.gateway.vpcId,
+          vpnGatewayId: this.target.gateway.routerTargetId,
+        });
+      }
```

Comment:
I know this will be done in next phases, but I want to keep track of the comments that will be done later to facilitate my review next time :) 

The attachment should be  moved to the place where we create the Gateways.

Reply:
Thanks @Mohamed, that's why i'm keeping this comment as unresolved so we could revisit this in next implementation along with adding in feature plan